numpy/_core/src/npysort/highway_qsort_16bit.dispatch.cpp

```diff
@@ -0,0 +1,20 @@
+#include "highway_qsort.hpp"
+#define VQSORT_ONLY_STATIC 1
+#include "hwy/contrib/sort/vqsort-inl.h"
+
+namespace np { namespace highway { namespace qsort_simd {
+
+template<> void NPY_CPU_DISPATCH_CURFX(QSort)(Half *arr, intptr_t size)
+{
+    hwy::HWY_NAMESPACE::VQSortStatic(reinterpret_cast<hwy::float16_t*>(arr), size, hwy::SortAscending());
+}
+template<> void NPY_CPU_DISPATCH_CURFX(QSort)(uint16_t *arr, intptr_t size)
```

Comment:
Just out of curiosity, do the uint16_t and int16_t versions also need `NPY_CPU_FEATURE_ASIMDHP`? I thought that was only required for vector float16 operations.